src/hera/workflows/_meta_mixins.py

```diff
@@ -764,9 +781,13 @@ def decorator(func: Callable[FuncIns, FuncR]) -> Callable:
             func_inputs = signature.parameters
             inputs = []
             if len(func_inputs) == 1:
-                arg_class = list(func_inputs.values())[0].annotation
-                if issubclass(arg_class, (InputV1, InputV2)):
-                    inputs = arg_class._get_inputs()
+                input_arg = list(func_inputs.values())[0].annotation
+                if issubclass(input_arg, (InputV1, InputV2)):
```

Comment:
Should we error if this isn't of correct type, rather than silently discarding it?